### PR TITLE
Fireeye feed

### PR DESCRIPTION
**Pull request by teizenman**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/24437

## Minimum version of Demisto
- [ ] 5.0.0
- [X] 5.5.0
- [ ] 6.0.0

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by ronykoz** [comment]
@teizenman ping me once it's ready for review

**Comment by teizenman** [comment]
@kirbles19 Can you please go over the following files:

- [ ] FeedFireEye_description.md
- [ ] README.md

**Comment by kirbles19** [comment]
@teizenman Done, no changes

File: Packs/FeedFireEye/Integrations/FeedFireEye/FeedFireEye.py [hunk 1]
```diff
@@ -0,0 +1,548 @@
...
+        relationships = dict()  # type: Dict
+        stix_entities = dict()  # type: Dict
+
+        headers = {
+            'Accept': 'application/vnd.oasis.stix+json; version=2.1',
+            'X-App-Name': 'content.xsoar.cortex.paloaltonetworks.v1.0',
+            'Authorization': f'Bearer {self.get_access_token()}'
+        }
+
+        if limit == -1:
```
[the remark this answers is not in the document]
Comment (teizenman) [reply Packs/FeedFireEye/Integrations/FeedFireEye/FeedFireEye.py:345]: Yes, the maximum limit available is 1000 by the API.
Added `max(limit, 1000)`  for situations in which the limit is defined

**Review by ronykoz — APPROVED** [review]
Let's do a demo before the merge

File: Packs/FeedFireEye/Integrations/FeedFireEye/FeedFireEye.py [hunk 2]
```diff
@@ -0,0 +1,563 @@
...
+        return auth_token
+
+    def get_access_token(self) -> str:
+        """Returns the current valid authentication token for the feed.
+
+        Returns:
+            str. Authentication token.
+        """
+        last_token_fetched_expiration_time = demisto.getIntegrationContext().get('expiration_time')
+        current_time = int(datetime.now().strftime('%s'))
```
Comment (roysagi) [inline Packs/FeedFireEye/Integrations/FeedFireEye/FeedFireEye.py:328]:
```suggestion
        current_time = int(datetime.now().timestamp())
```
isn't this what you are trying to do here?

File: Packs/FeedFireEye/Integrations/FeedFireEye/FeedFireEye.py [hunk 3]
```diff
@@ -0,0 +1,563 @@
...
+                timeout=self._polling_timeout,
+                resp_type='response'
+            )
+
+            if response.status_code == 204:
+                demisto.info(f'{INTEGRATION_NAME} info - '
+                             f'API Status Code: {response.status_code} No Content Available for this timeframe.')
+                return [], {}, {}
+
+            if response.status_code != 200:
```
Comment (roysagi) [inline Packs/FeedFireEye/Integrations/FeedFireEye/FeedFireEye.py:378]: shouldn't this return_error of some kind? if not how do we alert the user about a possible error? - same for `fetch_all_reports_from_api`

**Review by roysagi — COMMENTED** [review]
@teizenman - this looks good just had 2 questions - see my comments

File: Packs/FeedFireEye/Integrations/FeedFireEye/FeedFireEye.py [hunk 4]
```diff
@@ -88,6 +98,36 @@ def process_indicator_value(indicator_pattern_value: str) -> Tuple[List, List, D
...
+
+        Notes:
+            In case the (current_date - publishing date of the indicator) < reputation_interval, the highest score the
+            indicator can get is SUSPICIOUS.
+        """
+        current_date = datetime.now()
+        published_date = datetime.strptime(date, '%Y-%m-%dT%H:%M:%S.%fZ')
+
+        if current_date - published_date < timedelta(days=reputation_interval):
+            for score, threshold in FE_CONFIDENCE_TO_REPUTATION.items():
```
Comment (ronykoz) [inline Packs/FeedFireEye/Integrations/FeedFireEye/FeedFireEye.py:122]: the dict is not sorted, it might cause problems

**Review by ronykoz — COMMENTED** [review]
Not sure why do we need the reputation level param

**Comment by teizenman** [comment]
@kirbles19 

Two more instance parameters were added:
* Malicious Threshold
* Reputation Interval

https://github.com/demisto/content/blob/04cb8946ebd9ea99be29530295c38204d67b8c94/Packs/FeedFireEye/Integrations/FeedFireEye/FeedFireEye.yml#L77:L91

Can you please go over it?

**Comment by kirbles19** [comment]
> @kirbles19
> 
> Two more instance parameters were added:
> 
> * Malicious Threshold
> * Reputation Interval
> 
> https://github.com/demisto/content/blob/04cb8946ebd9ea99be29530295c38204d67b8c94/Packs/FeedFireEye/Integrations/FeedFireEye/FeedFireEye.yml#L77:L91
> 
> Can you please go over it?

Done

**Pull request merged**